soroban-env-host/src/host.rs

```diff
@@ -1895,10 +1895,50 @@ impl VmCallerEnv for Host {
 
         let min_expiration =
             self.with_ledger_info(|li| Ok(li.sequence_number.saturating_add(min.into())))?;
-        self.0.expiration_bumps.borrow_mut().0.push(LedgerBump {
-            key,
-            min_expiration,
-        });
+        self.0.expiration_bumps.borrow_mut().0.push_back(
+            LedgerBump {
+                key,
+                min_expiration,
+            },
+            self.as_budget(),
+        )?;
+        Ok(Val::VOID)
+    }
+
+    fn bump_current_contract_instance_and_code(
+        &self,
+        _vmcaller: &mut VmCaller<Host>,
+        min: U32Val,
+    ) -> Result<Void, HostError> {
+        let min_expiration =
+            self.with_ledger_info(|li| Ok(li.sequence_number.saturating_add(min.into())))?;
+
+        let contract_id = self.get_current_contract_id_internal()?;
+        let key = self.contract_instance_ledger_key(&contract_id)?;
+        self.0.expiration_bumps.borrow_mut().0.push_back(
```

Comment:
Does this actually work as expected? As far as I can see this operation is immutable.

Reply:
Yeah you are right... The fact that this is immutable makes it a lot harder to work with. 
I have reverted back to using a plain `Vec`.